draft-ietf-mls-architecture.md

```diff
@@ -795,6 +795,36 @@ If the Commit references an unknown proposal, group members may need to solicit
 the Delivery Service or other group members individually for the contents of the
 proposal.
 
+### Welcome Messages
+
+Whenever a commit adds new members to a group, MLS requires the committer to
+send a Welcome message to the new members. Applications should ensure that
+Welcome messages are coupled with the tie-breaking logic for commits, discussed
+in {{strongly-consistent}} and {{eventually-consistent}}. That is, when multiple
+commits are sent for the same epoch, applications need to ensure that only
+Welcome messages corresponding to the commit that "succeeded" are processed by
+new members.
+
+This is particularly important when groups are being reinitialized. When a group
+is reinitialized, it is restarted with a different protocol version and/or
+ciphersuite but identical membership. Whenever an authorized member sends and
+commits a ReInit proposal, this immediately freezes the existing group and
+triggers the creation of a new group with a new `group_id`.
+
+Ideally, the new group would be created by the same member that committed the
+`ReInit` proposal (including sending Welcome messages for the new group to all
+of the previous group's members). However this operation is not always atomic,
+so it's possible for a member to go offline after committing a ReInit proposal
+but before creating the new group. When this happens, it's necessary for another
+member to continue the reinitialization by creating the new group and sending
+out Welcome messages.
```

Comment:
Well, it is possible to design a client that does that, but it is also possible to design ones that always do generate the new group atomically. Is there a *technical* reason to NOT generate the ReInit Commit and create the new group as a single step?

Reply:
There's not a technical reason to make it not atomic. Like you say, it's possible, and doing so would avoid this issue. But I don't think it's something worth requiring